packages/api/src/providers/ActivitySendStatusTelemetry/ActivitySendStatusTelemetryComposer.tsx

```diff
@@ -0,0 +1,35 @@
+import { useGetActivityByKey, useSendStatusByActivityKey, useTrackEvent } from '../../hooks';
+
+import usePrevious from '../../hooks/internal/usePrevious';
+
+const ActivitySendStatusTelemetryComposer = () => {
+  const [activityToSendStatusMap] = useSendStatusByActivityKey();
+  const previousActivityToSendStatusMap = usePrevious(activityToSendStatusMap);
+  const getActivityByKey = useGetActivityByKey();
+  const trackEvent = useTrackEvent();
+
+  if (activityToSendStatusMap && previousActivityToSendStatusMap) {
+    const allActivityKeys = activityToSendStatusMap.keys();
+
+    for (const key of allActivityKeys) {
+      const currentStatus = activityToSendStatusMap.get(key);
+      const previousStatus = previousActivityToSendStatusMap.get(key);
+
+      if (!!currentStatus && !!previousStatus && currentStatus !== previousStatus) {
+        const activity = getActivityByKey(key);
+        const telemetryPayload = {
+          currentStatus: currentStatus.toString(),
+          previousStatus: previousStatus.toString(),
+          clientActivityID: activity?.channelData?.clientActivityID,
+          type: activity?.type?.toString(),
+          key,
+          hasAttachment: activity?.type === 'message' && activity?.attachments?.length > 0 ? 'true' : 'false'
+        };

```

Comment:
Notes:
- No need `toString()` because `type SendStatus extends string`, we don't use `enum`
   - Same to `activity.type.toString()`
- Sorting the object by alphabetical order of the key
- Saving `activity.type` so we save one member accessor
- If this is an event activity, `hasAttachment` should be `undefined` instead of `false`

```suggestion
        const type = activity?.type;
        const telemetryPayload: TelemetrySendStatusChangePayload = {
          clientActivityID: activity?.channelData?.clientActivityID,
          hasAttachment: type === 'message' && activity?.attachments?.length > 0 ? 'true' : 'false',
          key,
          prevStatus,
          status,
          type
        };
```

Reply:
updated